mypy/stubtest.py

```diff
@@ -216,18 +220,18 @@ def verify_mypyfile(
         m
         for m in dir(runtime)
         if not m.startswith("_")
-        # Ensure that the object's module is `runtime`, e.g. so that we don't pick up reexported
-        # modules and infinitely recurse. Unfortunately, there's no way to detect an explicit
-        # reexport missing from the stubs (that isn't specified in __all__)
-        and getattr(getattr(runtime, m), "__module__", None) == runtime.__name__
     ]
     # Check all things declared in module's __all__, falling back to runtime_public_contents
     to_check.update(getattr(runtime, "__all__", runtime_public_contents))
```

Comment:
I believe this is incorrect, you just have to updated with both `__all__` and `runtime_public_contents`. 

`__all__` is not about things being public or private, it is just about `from pgk import *`. For example, you could have tons of public stuff in a package, but just list just the most useful things in `__all__`.

However, consider that `__all__` can list names starting with understcore (`_`), not sure if those should be considered public or private in mypy, but `from pgk import *` do import them.

Reply:
Type checkers use `__all__` as a way to determine the public API:

https://mail.python.org/archives/list/typing-sig@python.org/message/FPTLFORIBPKY36UO7PLO47HJDCLG2KD5/ https://github.com/python/mypy/blob/5aa707d8223c3be8d52b26e444f3b8c67f50d2f2/mypy/semanal.py#L483

Note that due to the logic on L218 (on the right), there's a class of stuff that stubtest will check if it's present in the stub and runtime, but won't complain about if it's missing from the stub.